leases/utils.py

```diff
@@ -43,6 +45,52 @@ def calculate_season_end_date(lease_end: date = None) -> date:
     return default
 
 
+def calculate_winter_season_start_date(lease_start: date = None) -> date:
+    """Return the date when the winter season starts
+
+    If the current date is after the season end date,
+    it returns next year's default start date.
+
+    Winter Storage Leases always start on 15.9 the earliest
+    """
+    today = date.today()
+    default = date(day=15, month=9, year=today.year)
+
+    if lease_start:
+        return default.replace(year=lease_start.year)
+
+    # If the current day is before the winter season ends,
+    # the start day will be on the previous year
+    if today < date(day=10, month=6, year=today.year):
+        default -= relativedelta(years=1)
+    # Otherwise, if the current day is after the season has started,
+    # the default start date will be on the following year
+    elif today >= date(day=15, month=9, year=today.year):
+        default += relativedelta(years=1)
+
+    return default
+
+
+def calculate_winter_season_end_date(lease_end: date = None) -> date:
+    """Return the date when the winter season ends
+
+    Winter Storage Leases always end on 10.6, the year depends on the current year
+    """
+    today = date.today()
+    default = date(day=10, month=6, year=today.year)
+
+    if lease_end:
+        return default.replace(year=lease_end.year)
```

Comment:
and here if the winter storage ends from 15.9 till 31.12, we should return 10.6 of the next year